app/models/simulator.rb

```diff
@@ -360,4 +360,29 @@ def simulator_versions
 
     sim_versions.map {|key,val| val['version'] = key; val }.sort_by {|a| a['latest_started_at']}
   end
+
+  require 'csv'
+  def runs_csv(runs = self.runs)
+    run_attr = %w(_id status hostname real_time started_at finished_at)
+    ps_attr = [[:ps,:_id]] + parameter_definitions.map {|pd| [:ps,:v,pd.key] }
+    ps_attr_header = ["ps_id"] + parameter_definitions.map {|pd| "p.#{pd.key}" }
+    latest_run = Run.where(simulator: self, status: :finished).order_by(updated_at: :desc).first
+    result_attr_header = plottable_keys(latest_run&.result).map {|key| "r.#{key}" }
```

Comment:
What is `latest_run&.result`?

Reply:
It is equivalent to `latest_run.try(:result)`. This syntax is introduced in Ruby 2.3.

Since `latest_run` can be nil, `&.` is necessary otherwise it may throw `NoMethodError`.